### PR TITLE
v3 Removal and Request Rework

### PR DESCRIPTION
**Pull request by swiftyspiffy**
summon big PR!

### CODE REVIEW

**Comment by LuckyNoS7evin** [comment]
Looks good after some of the changes you made

File: TwitchLib.Api/TwitchAPI.cs [hunk 1]
```diff
@@ -21,9 +21,12 @@ public class TwitchAPI : ITwitchAPI
         private readonly HttpClient _http;
         private readonly TwitchLibJsonSerializer _jsonSerializer;
         private readonly IRateLimiter _rateLimiter;
+
+        internal const string baseV5 = "https://api.twitch.tv/kraken";
```
Comment (Syzuna) [inline TwitchLib.Api/TwitchAPI.cs]: still lower case :P

File: TwitchLib.Api/TwitchAPI.cs [hunk 2]
```diff
@@ -21,9 +21,12 @@ public class TwitchAPI : ITwitchAPI
...
+        internal const string baseHelix = "https://api.twitch.tv/helix";
```
Comment (Syzuna) [inline TwitchLib.Api/TwitchAPI.cs]: still lower case :P

File: TwitchLib.Api/TwitchAPI.cs [hunk 3]
```diff
@@ -97,49 +95,93 @@ public TwitchAPI(ILogger<TwitchAPI> logger = null, IRateLimiter rateLimiter = nu
...
-        #region Post
-        internal async Task<KeyValuePair<int, string>> PostAsync(string url, string payload, List<KeyValuePair<string, string>> getParams = null, string accessToken = null, ApiVersion api = ApiVersion.v5, string clientId = null)
+
+        internal async Task<KeyValuePair<int, string>> TwitchPostAsync(string resource, ApiVersion api, string payload, List<KeyValuePair<string, string>> getParams = null, string accessToken = null, string clientId = null, string customBase = null)
         {
+            string url = constructResourceUrl(resource, getParams, api, customBase);
+            return await _rateLimiter.Perform(async () => await GeneralRequestAsync(url, HttpMethod.Post, payload, accessToken, api, clientId));
+        }
+
+        private string constructResourceUrl(string resource = null, List<KeyValuePair<string, string>> getParams = null, ApiVersion api = ApiVersion.v5, string overrideUrl = null)
```
Comment (Syzuna) [inline TwitchLib.Api/TwitchAPI.cs]: lower case method name NotLikeThis (Please dont hate me :D)

**Pull request merged**